rules/DeadCode/Rector/If_/RemoveDeadInstanceOfRector.php

```diff
@@ -146,4 +145,26 @@ private function isPropertyFetch(Expr $expr): bool
 
         return $expr instanceof StaticPropertyFetch;
     }
+
+    private function isInstanceofTheSameType(Instanceof_ $instanceof): ?bool
+    {
+        if (! $instanceof->class instanceof Name) {
+            return null;
+        }
+
+        // handled in another rule
+        if ($this->isPropertyFetch($instanceof->expr) || $instanceof->expr instanceof CallLike) {
+            return null;
+        }
+
+        $classType = $this->nodeTypeResolver->getType($instanceof->class);
+        $exprType = $this->nodeTypeResolver->getType($instanceof->expr);
```

Comment:
`getNativeType()` should be used to ensure not change type from docblock

```php
/** 
 * @param Node $var
 */
public function go($var)
    {
        if ($var instanceof Node && $var->getLine() === 100) {
        }
    }
  ```

Reply:
Oh, I see, it was here from the beginning, so it seems ok, from docblock should be covered in separate PR, if needed ;)